src/depthquality/quality.py

```diff
@@ -87,6 +91,15 @@ def clip_pointcloud_to_pattern_area(reference_mesh, aligned_pointcloud, depth_sc
 
     return cropped_pointcloud
 
+def compute_corner_coordinates(pointcloud, camera_matrix, corner_coordinates):
+
+    for point in np.asarray(pointcloud.points):
+        u = np.floor(camera_matrix["fx"] * point[0] / point[2] + camera_matrix["ppx"])
+        v = np.floor(camera_matrix["fy"] * point[1] / point[2] + camera_matrix["ppy"])
+
+        if (v, u) in corner_coordinates.keys():
+            corner_coordinates[(v,u)] = point
```

Comment:
Here you are overwriting the 3D point every time you see a new projected point.

For much higher accuracy, I would append `point` to the list (since that's what you're passing in anyway) and take the mean at the end of this function before returning.

Reply:
Made this change in the latest commit.